dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,546 @@
+        self.best_estimator_ = best_estimator
+        self.best_score_ = scores[best_bracket]
+        self.best_params_ = cv_results["params"][best_index]
+        self.scorer_ = scorer
+
+        self.estimator_history_ = estimator_history
+        self.history_ = history
+        self.cv_results_ = cv_results
+
+        self.multimetric_ = SHAs[best_bracket].multimetric_
+        raise gen.Return(self)
+
+    def metadata(self):
+        """Get information about how much computation is required for
+        :func:`~dask_ml.model_selection.HyperbandSearchCV.fit`. This can be called
+        before or after  :func:`~dask_ml.model_selection.HyperbandSearchCV.fit`.
+
+        Returns
+        -------
+        metadata : dict
+            Information about the computation performed by ``fit``. Has keys
+
+            * ``partial_fit_calls``, the total number of partial fit calls.
+            * ``models``, the total number of models created.
+            * ``brackets``, each of which has the same two keys as above.
+
+        Notes
+        ------
+        This algorithm runs several loops in an "embarassingly parallel"
+        manner. The ``brackets`` key represents each of these loops.
+
+        """
+        bracket_info = _hyperband_paper_alg(self.max_iter, eta=self.aggressiveness)
+        num_models = sum(b["models"] for b in bracket_info)
+        for bracket in bracket_info:
+            bracket["iters"].update({1})
+            bracket["iters"] = sorted(list(bracket["iters"]))
+        num_partial_fit = sum(b["partial_fit_calls"] for b in bracket_info)
+        bracket_info = list(reversed(sorted(bracket_info, key=lambda x: x["bracket"])))
+
+        N, R, brackets = _get_hyperband_params(self.max_iter, eta=self.aggressiveness)
+        SHAs = {bracket: SHA for bracket, SHA in self._get_SHAs(N, R, brackets)}
+        for bracket in bracket_info:
+            b = bracket["bracket"]
+            bracket["SuccessiveHalvingSearchCV params"] = _get_SHA_params(SHAs[b])
+
+        info = {
+            "partial_fit_calls": num_partial_fit,
+            "models": num_models,
+            "brackets": {"bracket=" + str(b["bracket"]): b for b in bracket_info},
+        }
+        return info
+
+
+def _get_meta(hists, brackets, SHAs, key=None):
+    if key is None:
+        key = lambda bracket, ident: "bracket={}-{}".format(bracket, ident)
+    meta_ = {}
+    history_ = {}
+    for bracket in brackets:
+        hist = hists[bracket]
+
+        info_hist = {key(bracket, h["estimator_id"]): [] for h in hist}
+        for h in hist:
+            info_hist[key(bracket, h["estimator_id"])] += [h]
+        hist = info_hist
+        history_.update(hist)
+
+        calls = {k: max(hi["partial_fit_calls"] for hi in h) for k, h in hist.items()}
+        iters = {hi["partial_fit_calls"] for h in hist.values() for hi in h}
+        meta_["bracket=" + str(bracket)] = {
+            "iters": sorted(list(iters)),
+            "models": len(hist),
```

Comment:
Wording: `models` to `estimators`.

Wording: `iters`

Reply:
`unique_partial_fit_calls` works instead of `iters`.

Reply:
Maybe `decisions` instead of `iters`? `9 in decisions` would mean "was a decision made just after 9 partial_fit calls?"